File: backend_java/src/main/java/br/com/fiap/chronosdtn/dto/TokenResponse.java
package br.com.fiap.chronosdtn.dto;

// Por que: Imutabilidade do token de retorno em chamadas de autenticação bem-sucedidas.
public record TokenResponse(
    // Por que: String contendo o token JWT criptografado gerado pela aplicação.
    String token,

    // Por que: Tipo de autenticação, fixo como Bearer (padrão RFC 6750).
    String type
) {
}

File: backend_net/ChronosDtnNetApi/DTOs/TokenResponse.cs
// Por que: Define o namespace DTOs para organizar a estrutura do JSON de retorno das chamadas de API.
namespace ChronosDtnNetApi.DTOs;

// Por que: Record imutável que carrega as credenciais em formato JWT para o cliente autenticado.
public record TokenResponse(
    // Por que: String contendo o token assinado gerado pelo AuthService.
    string Token,

    // Por que: Define o tipo do token, padrão da indústria "Bearer".
    string Type
);
